### DISCUSSION TITLE
What is the plan for discovering photos (search, tags, albums)?

### DISCUSSION THREAD

**Discussion by bnigh** [immich-app/immich#1310] [General]
From discord and various github issues, it sounds like there is ongoing effort to refactor the tagging system and search (web (#1021)). Since there are a number of github issues related to search, tagging, and album management, I thought it may be better to pose this question / start a discussion in a new thread.

### Questions

Given that discovery is such a large scope, what is the plan for the current refactoring?

Beyond the current refactoring, what is the vision for the various ways a user may discover photos?
- for example:
  - via scrolling through the timeline
  - via a manually created album
  - via a tag (either applied manually or by ai... this may have some overlap with albums) (related to https://github.com/immich-app/immich/issues/838)
  - via search / filtering (related to #1021)
  - via global map (related to #238)

Specifically for albums / tags, as a user's data set scales, how will they discover what tags / albums are available to choose from?
- for example:
  - via scrolling through available albums / tags
  - via some nested / hierarchical structure (related to #1010)
  - via search / filter (related to #1008)

Specifically for search,
- what attributes will users be able to utilize in the search?
- are there future plans for queries using logical operators?
  - for example:
    - If I have different metadata (tags) or albums for 2 pets (`pets/max`, `pets/luna`), would it be possible to search for photos only containing bother max and luna (photos with tags for both `pets/max` and `pets/luna` / photos belonging to albums for both `pets/max`, `pets/luna`)

### Motivating Use Cases

For my personal photos, I haven't yet curated with appropriate metadata, but the following are some different patterns I could imagine for discovering / finding a particular photo or set of photos:

- I would like to find photos by a specific date / date range
  - the existing timeline gives an excellent experience for this
- I would like to find photos by a specific event (ie: `2022 Thanksgiving`)
  - the `event` would need to be manually added metadata (tag / album)
- I would like to find photos by person / pet
  - the `person` / `pet` metadata could be added manually, by ai, or by a mix of both
- I would like to find photos by location
- I would like to find photos by thing (object / items / activity) within the photo
  - this metadata could be added manually, by ai, or by a mix of both

**Comment by dfb** [comment]
AI-based automatic annotation is becoming doable, and in theory could deliver a lot of the motivating use cases, see e.g. https://jiahaoli.net/omniquery/
